main/registerAutoUpdaterListeners.ts

```diff
@@ -21,34 +23,44 @@ export default function registerAutoUpdaterListeners(main: Main) {
     emitMainProcessError(error);
   });
 
-  // eslint-disable-next-line @typescript-eslint/no-misused-promises
-  autoUpdater.on('update-available', async (info: UpdateInfo) => {
-    const currentVersion = app.getVersion();
-    const nextVersion = info.version;
-    const isCurrentBeta = currentVersion.includes('beta');
-    const isNextBeta = nextVersion.includes('beta');
-
-    let downloadUpdate = true;
-    if (!isCurrentBeta && isNextBeta) {
-      const option = await dialog.showMessageBox({
-        type: 'info',
-        title: 'Update Available',
-        message: `Download version ${nextVersion}?`,
-        buttons: ['Yes', 'No'],
-      });
-
-      downloadUpdate = option.response === 0;
-    }
+  autoUpdater.on('update-available', (info: UpdateInfo) => {
+    void (async () => {
+      const updateInfo = info;
 
-    if (!downloadUpdate) {
-      return;
-    }
+      availableUpdate = updateInfo;
+
+      const currentVersion = app.getVersion();
+      const nextVersion = updateInfo.version;
+      const isCurrentBeta = currentVersion.includes('beta');
+      const isNextBeta = nextVersion.includes('beta');
+
+      let downloadUpdate = true;
+      if (!isCurrentBeta && isNextBeta) {
+        const option = await dialog.showMessageBox({
+          type: 'info',
+          title: 'Update Available',
+          message: `Download version ${nextVersion}?`,
+          buttons: ['Yes', 'No'],
```

Comment:
Haven't checked out how this looks in the Ui. If you could attach a screenshot that would be helpful :). 

If the design permits rendering a longer message, Please convey the message what actions are performed if user proceeds. 

The title wrote here is `Update Available`. But the description only mentions about downloading a version. So, from a user POV,  I'm slightly confused, proceeding with Yes will download the latest version and update the software right ?. If so that should be mentioned clearly. 

Also it would be helpful to notify the user, taking a DB Backup is encouraged.